### ISSUE TITLE
Automatically clean up `QubesIncoming` directory if empty

### ISSUE THREAD

**Issue by andrewdavidwong** [QubesOS/qubes-issues#8307]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### The problem you're addressing (if any)

After copying or moving a file from one qube to another, a `QubesIncoming` directory is created along with suitable subdirectories. Usually, the user moves the file out of this directory structure in order to make use of the file. This often leaves behind an empty `QubesIncoming` directory (often with empty subdirectories). These persist until the user manually cleans them up.

### The solution you'd like

Automatically remove the empty `QubesIncoming` directory whenever the qube starts up or shuts down (not sure which is better). For present purposes:

- A `QubesIncoming` directory that contains only empty subdirectories still counts as empty.
- A `QubesIncoming` directory that contains non-default hidden files does not count as empty.

### The value to a user, and who that user might be

Less clutter. Less time and effort spent manually cleaning up after the system.

**Comment by PHCEAC** [comment]
Will this be enabled by default? Or has it been?

There is useful data/metadata in the names and modification times of those directories. Could even be very useful if a VM got compromised... to determine we're it had touched.

(I was surprised by the lack of an "Incoming" directory recently, but I attributed it to PEBKAC. Clearly, I really need logging of qvm-copy activity, and I did not yet look into it.)
